public/app/features/alerting/unified/components/mute-timings/MuteTimingTimeInterval.tsx

```diff
@@ -22,114 +25,122 @@ export const MuteTimingTimeInterval = () => {
+                    defaultValue={{ label: timeInterval.location, value: timeInterval.location }}
+                    data-testid="mute-timing-location"
+                  />
+                </Field>
+                <Field label="Days of the week">
+                  <DaysOfTheWeek
+                    onChange={(daysOfWeek) => {
+                      setValue(`time_intervals.${timeIntervalIndex}.weekdays`, daysOfWeek);
+                    }}
+                    // @ts-ignore react-hook-form doesn't handle nested field arrays well
+                    defaultValue={timeInterval.weekdays}
+                  />
+                </Field>
+                <Field
+                  label="Days of the month"
+                  description="The days of the month, 1-31, of a month. Negative values can be used to represent days which begin at the end of the month"
+                  invalid={!!errors.time_intervals?.[timeIntervalIndex]?.days_of_month}
+                  error={errors.time_intervals?.[timeIntervalIndex]?.days_of_month?.message}
+                >
+                  <Input
+                    {...register(`time_intervals.${timeIntervalIndex}.days_of_month`, {
+                      validate: (value) =>
+                        validateArrayField(
+                          value,
+                          (day) => {
+                            const parsedDay = parseInt(day, 10);
+                            return (parsedDay > -31 && parsedDay < 0) || (parsedDay > 0 && parsedDay < 32);
+                          },
+                          'Invalid day'
+                        ),
+                    })}
+                    width={50}
+                    // @ts-ignore react-hook-form doesn't handle nested field arrays well
+                    defaultValue={timeInterval.days_of_month}
+                    placeholder="Example: 1, 14:16, -1"
+                    data-testid="mute-timing-days"
+                  />
+                </Field>
+                <Field
+                  label="Months"
+                  description="The months of the year in either numerical or the full calendar month"
+                  invalid={!!errors.time_intervals?.[timeIntervalIndex]?.months}
+                  error={errors.time_intervals?.[timeIntervalIndex]?.months?.message}
+                >
+                  <Input
+                    {...register(`time_intervals.${timeIntervalIndex}.months`, {
+                      validate: (value) =>
+                        validateArrayField(
+                          value,
+                          (month) => MONTHS.includes(month) || (parseInt(month, 10) < 13 && parseInt(month, 10) > 0),
+                          'Invalid month'
+                        ),
+                    })}
+                    width={50}
+                    placeholder="Example: 1:3, may:august, december"
+                    // @ts-ignore react-hook-form doesn't handle nested field arrays well
```

Comment:
You'll see `// @ts-ignore` in a few places – this basically comes down to `react-hook-form` being sort of clueless about the types of nested fields when using `useFieldArray<>()`